src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -220,6 +226,17 @@ public AzureAppConfigurationOptions Select(string keyFilter, string labelFilter
                 labelFilter = LabelFilter.Null;
             }
 
+            if (tagsFilter != null)
+            {
+                foreach (var tag in tagsFilter)
+                {
+                    if (string.IsNullOrEmpty(tag) || !tag.Contains('=') || tag.IndexOf('=') == 0)
```

Comment:
Does SDK throw any error if the tag filter is malformed?

Reply:
Yes, a RequestFailedException that will say invalid parameter for tags. This also changes the optional behavior, is it better to just not check the format here?

Reply:
So it sounds like this error is coming from the backend. SDK is just relaying the error as a RequestFailedException. It's a bad input error so we should be okay to throw it even if optional is true. I think its ok to have basic validation here to fail fast, just like we validate key and label filters above. All possible scenarios will not be covered here, but checking for '=' sign and non-empty tag name is a good basic check.